### PR TITLE
Added helper function to calculate optimal control variate coefficient

### PR DESCRIPTION
**Pull request by lbollar**
WIP 
Referencing [#4](https://github.com/blei-lab/edward/issues/4), this is an attempt to create a helper function for implementing a vanilla control variates method for variance reduction.

Right now, I just have this calculating the coefficient as this is what I thought was really needed to use this technique.

I tested this function with following implementation in build_score_loss_and_gradients in klqp.py: 
```
  f = q_log_prob * losses
  h = q_log_prob

  a = get_control_variate_scalar(f, h)

  grads = tf.gradients(
      -tf.reduce_mean(q_log_prob * tf.stop_gradient(losses - a)),
      [v._ref() for v in var_list])
```

Using this on the [Mixture Model](http://edwardlib.org/tutorials/unsupervised) example, this seemed to converge in under half the the time of original implementation and was achieving a loss of under 1000 in under 400 iterations. 

Apologies for latex errors that might exist, that is not my strong point.

Is this the right direction?

### CODE REVIEW

**Comment by dustinvtran** [comment]
Neat. The implementation is clear. Moving it into `edward/util/tensorflow.py` seems like the best place for now until we build up more control variates. 

Can you also add a unit test? You can use similar tests such as [`test_reduce_logmeanexp.py`](https://github.com/blei-lab/edward/blob/master/tests/test-util/test_reduce_logmeanexp.py) as reference. To run tests, you'll need to install `py.test`. This lets you call, e.g., `py.test test_reduce_logmeanexp.py`.

Once we merge this, we can go ahead and start adding it as an option for the black box methods.

**Comment by dustinvtran** [comment]
Great! Lastly, I think you need to add `get_control_variate_scalar` to `edward/__init__.py`, so we can call `ed.get_control_variate_scalar`.

**Comment by dustinvtran** [comment]
Looks like there's a merge conflict. You'll have to merge/rebase to master; also I'd prefer it if it was added to __init__ in alphabetical order :)

**Comment by lbollar** [comment]
Sorry, lack of attention to detail on my part :)
Now done.
